README.md

```diff
@@ -1,2 +1,230 @@
-# fairing
-An experimental library to fire of TFJob, PyTorchJob etc... from Jupyter
+# Fairing
+
+Easily train and serve ML models on Kubernetes, directly from your python code.  
+
+This projects uses [Metaparticle](http://metaparticle.io/) behind the scene.
+
+fairing allows you to express how you want your model to be trained and served using native python decorators.  
+
+
+## Table of Contents
+
+- [Requirements](#requirements)
+- [Getting `fairing`](#getting-fairing)
+- [Training](#training)
+  - [Simple Training](#simple-training)
+  - [Hyperparameters Tuning](#hyperparameters-tuning)
+  - [Population Based Training](#population-based-training)
+- [Usage with Kubeflow](#usage-with-kubeflow)
+  - [Simple TfJob](#simple-tfjob)
+  - [Distributed Training](#distributed-training)
+  - [From a Jupyter Notebook](#from-a-jupyter-notebook)
+- [Monitoring with TensorBoard](#tensorboard)
+
+## Requirements
+
+If you are going to use `fairing` on your local machine (as opposed to from a Jupyter Notebook deployed inside a Kubernetes cluster for example), you will need 
+to have access to a deployed Kubernetes cluster, and have the `kubeconfig` for this cluster on your machine.
+
+You will also need to have docker installed locally.
+
+## Getting `fairing`
+
+**Note**: This projects requires python 3
+
+```bash
+pip install fairing
+```
+
+Or, in a Jupyter Notebook, create a new cell and execute: `!pip install fairing`.
+
+## Training
+
+`fairing` provides a `@Train` class decorator allowing you to specify how you want your model to be packaged and trained.  
+Your model needs to be defined as a class to work with `fairing`. 
+
+This limitation is needed in order to enable usage of more complex training strategies and simplify usage from within a Jupyter Notebook.
+
+Following are a series of example that should help you understand how fairing works.
+<!-- The train decorator 
+* `package`: Defines the repository (this could be your DockerHub username, or something like `somerepo.acr.io` on Azure for example) and name that should be used to build the image. You can control wether you want to publish the image by setting `publish` to `True`.
+* `strategy`: Specify which training strategy should be used (more details below).
+* `architecture`: Specify which architecture should be used. (more details below)
+* `tensorboard`: [Optional] If specified, will spawn an instance of TensorBoard to monitor your trainings
```

Comment:
Should this be the responsibility of `fairing`? (Not a leading question, I'm still understanding the whole project).